http-client-jdk/src/main/java/io/micronaut/http/client/jdk/DefaultJdkHttpClientRegistry.java

```diff
@@ -0,0 +1,328 @@
+import io.micronaut.inject.qualifiers.Qualifiers;
+import io.micronaut.json.JsonFeatures;
+import io.micronaut.json.JsonMapper;
+import io.micronaut.json.codec.MapperMediaTypeCodec;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Factory to create {@literal java.net.http.*} HTTP Clients.
+ *
+ * @author Sergio del Amo
+ * @author Tim Yates
+ * @since 4.0.0
+ */
+@Factory
+@BootstrapContextCompatible
+@Internal
+@Experimental
+public final class DefaultJdkHttpClientRegistry implements AutoCloseable, HttpClientRegistry<HttpClient> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(DefaultJdkHttpClientRegistry.class);
+
+    private final Map<ClientKey, HttpClient> clients = new ConcurrentHashMap<>(10);
+    private final BeanContext beanContext;
+    private final LoadBalancerResolver loadBalancerResolver;
+    private final HttpClientConfiguration defaultHttpClientConfiguration;
+    private final JsonMapper jsonMapper;
+    @Nullable
+    private final MediaTypeCodecRegistry mediaTypeCodecRegistry;
+    private final BeanProvider<RequestBinderRegistry> requestBinderRegistryProvider;
+    private final JdkClientSslBuilder jdkClientSslBuilder;
+
+    public DefaultJdkHttpClientRegistry(
+        BeanContext beanContext,
+        LoadBalancerResolver loadBalancerResolver, HttpClientConfiguration defaultHttpClientConfiguration,
+        JsonMapper jsonMapper,
+        @Nullable MediaTypeCodecRegistry mediaTypeCodecRegistry,
+        BeanProvider<RequestBinderRegistry> requestBinderRegistryProvider,
+        BeanProvider<JdkClientSslBuilder> sslBuilderBeanProvider
+    ) {
+        this.beanContext = beanContext;
+        this.loadBalancerResolver = loadBalancerResolver;
+        this.defaultHttpClientConfiguration = defaultHttpClientConfiguration;
+        this.jsonMapper = jsonMapper;
+        this.mediaTypeCodecRegistry = mediaTypeCodecRegistry;
+        this.requestBinderRegistryProvider = requestBinderRegistryProvider;
+        this.jdkClientSslBuilder = sslBuilderBeanProvider.orElse(new JdkClientSslBuilder(new ResourceResolver()));
+    }
+
+    private static MediaTypeCodec createNewJsonCodec(BeanContext beanContext, JsonFeatures jsonFeatures) {
+        return getJsonCodec(beanContext).cloneWithFeatures(jsonFeatures);
+    }
+
+    private static MapperMediaTypeCodec getJsonCodec(BeanContext beanContext) {
+        return beanContext.getBean(MapperMediaTypeCodec.class, Qualifiers.byName(MapperMediaTypeCodec.REGULAR_JSON_MEDIA_TYPE_CODEC_NAME));
+    }
+
+    /**
+     * Creates a {@literal java.net.http.*} HTTP Client.
+     *
+     * @param injectionPoint
+     * @param loadBalancer
+     * @param configuration
+     * @param beanContext
+     * @return A {@literal java.net.http.*} HTTP Client
+     */
+    @Bean
+    @BootstrapContextCompatible
+    @Primary
```

Comment:
what happens if both clients on the classpath? Perhaps `@Order` should be used here to define priority

Reply:
👍